packages/filesystem/src/browser/download/file-download-service.ts

```diff
@@ -78,8 +79,15 @@ export class FileDownloadService {
                 progress.cancel();
                 const downloadUrl = `${this.endpoint()}/download/?id=${jsonResponse.id}`;
                 if (copyLink) {
+                    let disposable: Disposable | undefined = undefined;
                     if (document.documentElement) {
-                        addClipboardListener(document.documentElement, 'copy', e => this.handleCopy(e, downloadUrl));
+                        try {
+                            disposable = addClipboardListener(document.documentElement, 'copy', e => this.handleCopy(e, downloadUrl));
+                        } finally {
+                            if (disposable) {
+                                disposable.dispose();
```

Comment:
it should be disposed within event handler before calling `this.handleCopy`.

Reply:
Can you explain why?

Reply:
because we want to wait till `copy` event is fired and remove a listener only after it? I don't understand how current change ensures it.

We use the same pattern in other places as well: https://github.com/theia-ide/theia/blob/60dcb3326ea5e70c8d1017dcd206013304a5c92e/packages/markers/src/browser/problem/problem-contribution.ts#L158-L168